### ISSUE TITLE
slide element is removed when bounded data gets changed

### ISSUE THREAD

**Issue by avivr** [devmark/angular-slick-carousel#27]
Changing a property on a data item which is bound to slick-slide, cause the slide to be deleted.

for example:

js:
```
    var view1 = { text: "My View1", property: 0 };
    var view2 = { text: "My View2", property: 0 };
    
    $rootScope.viewData = [view1, view2];
```

html:
```
<slick data="viewData">
      <div ng-repeat="view in viewData">
          <div>{{view.text}}</div>
      </div>
</slick>
```

when changing view1.property value, the slide element gets deleted from DOM.

Here is a plunker with an example: 
http://plnkr.co/edit/iDffFOpq5s2Zmpj9q40f?p=preview

you can press the 'update view' button to make 'view*.property' change and see that the slides get removed.
click the button several times to see how it impact all slides.

**Comment by devmark** [comment]
Hello, 
will fix this today. thanks

**Comment by devmark** [comment]
updated. but i will make more stable for that. for workaround  you can use version 3.0.3

**Comment by avivr** [comment]
@devmark  i'm now getting:

```
TypeError: Cannot read property 'length' of undefined
    at Object.fn (angular-slick.min.js:8)
    at Scope.$digest (angular.js:15683)
    at Scope.$apply (angular.js:15951)
    at done (angular.js:10364)
    at completeRequest (angular.js:10536)
    at XMLHttpRequest.requestLoaded (angular.js:10477)
```

you shouldn't use 'length' to compare.. and also greater-than (>) is not good enough

**Comment by devmark** [comment]
3.0.4

**Comment by avivr** [comment]
the bug is not fixed in 3.0.4

**Comment by devmark** [comment]
check 3.0.5
